packages/theme-product/theme-product.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Find a match in the project JSON (using a ID number) and return the variant (as an Object)
+ * @param {Object} product Product JSON object
+ * @param {Number} value Accepts Number (e.g. 6908023078973)
+ * @returns {Object} The variant object once a match has been successful. Otherwise null will be return
+ */
+export function getVariantFromId(product, value) {
+  _validateProductStructure(product);
+
+  if (typeof value === 'number') {
+    var result = product.variants.filter(function(variant) {
+      return variant.id === value;
+    });
+
+    return result[0] || null;
+  }
+
+  return null;
+}
+
+/**
+ * Convert the Object (with 'name' and 'value' keys) into an Array of values, then find a match & return the variant (as an Object)
+ * @param {Object} product Product JSON object
+ * @param {Object} collection Object with 'name' and 'value' keys (e.g. [{ name: "Size", value: "36" }, { name: "Color", value: "Black" }])
+ * @returns {Object} The variant object once a match has been successful. Otherwise null will be returned
+ */
+export function getVariantFromSerializedArray(product, collection) {
+  _validateProductStructure(product);
+
+  // If value is an array of options
+  var optionArray = _createOptionArrayFromOptionCollection(product, collection);
+  return getVariantFromOptionArray(product, optionArray);
+}
+
+/**
+ * Find a match in the project JSON (using Array with option values) and return the variant (as an Object)
+ * @param {Object} product Product JSON object
+ * @param {Array} options List of submitted values (e.g. ['36', 'Black'])
+ * @returns {Object} The variant object once a match has been successful. Otherwise an empty object will be returned
+ */
+export function getVariantFromOptionArray(product, options) {
+  _validateProductStructure(product);
+  _validateOptionsArray(options);
+
+  var result = product.variants.filter(function(variant) {
+    return options.every(function(option, index) {
+      return variant.options[index] === option;
+    });
+  });
+
+  return result[0] || null;
+}
+
+/**
+ * Creates an array of selected options from the object
+ * Loops through the project.options and check if the "option name" exist (product.options.name) and matches the target
+ * @param {Object} product Product JSON object
+ * @param {Array} collection Array of object (e.g. [{ name: "Size", value: "36" }, { name: "Color", value: "Black" }])
+ * @returns {Array} The result of the matched values. (e.g. ['36', 'Black'])
+ */
+function _createOptionArrayFromOptionCollection(product, collection) {
+  _validateProductStructure(product);
+  _validateSerializedArray(collection);
+
+  var optionArray = [];
+  var indexOption = -1;
+
+  collection.forEach(function(option) {
+    if (typeof option.name !== 'string') {
```

Comment:
This check could be part of `_validateSerializedArray`

Reply:
I added in `_validateSerializedArray()`. To check if the property exists without going through a loop, I used the `hasOwnProperty` method.